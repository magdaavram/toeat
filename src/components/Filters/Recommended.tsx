import React, { useEffect, useState } from 'react';
import Tag from 'components/Tag';
import styled from 'styled-components';
import getRecommendedFilters, { IFilter } from 'api/filters';

const Container = styled.div`
  width: 70%;
  max-width: 1000px;
  margin: 18px;
  background-color: rgba(201, 173, 167, 0.3);
  padding: 18px;
  border-radius: 12px;
`;

const Header = styled.h3`
  padding: 0;
  margin: 0 0 18px 0;
  color: var(--color--light-purple);
  font-size: var(--font-size--large);
`;

const TagsContainer = styled.div`
  flex: flex wrap;
`;

const Recommended = () => {
  const [filters, setFilters] = useState<IFilter[]>([]);

  useEffect(() => {
    setFilters(getRecommendedFilters());
  }, []);

  const filtersList = filters.map((filter: IFilter, index) => (
    <Tag
      name={filter.name}
      active={filter.active}
      key={`filter-${index}`}
      onClick={() => console.log(`pressed ${filter.name}`)}
    />
  ));

  return (
    <Container>
      <Header>Recommended ingredients</Header>
      <TagsContainer>{filtersList}</TagsContainer>
    </Container>
  );
};

export default Recommended;
