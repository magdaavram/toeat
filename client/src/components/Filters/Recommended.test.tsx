import React from 'react';
import { mount } from 'enzyme';
import Recommended from './Recommended';

it('should render correctly', () => {
  const wrapper = mount(<Recommended />);

  expect(wrapper).toMatchSnapshot();
});
