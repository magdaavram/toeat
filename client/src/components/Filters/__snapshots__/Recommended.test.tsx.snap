// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<Recommended>
  <styled.div>
    <div
      className="sc-AxirZ xjolV"
    >
      <styled.h3>
        <h3
          className="sc-AxiKw gUYhqW"
        >
          Recommended ingredients
        </h3>
      </styled.h3>
      <styled.div>
        <div
          className="sc-AxhCb bxdZFX"
        >
          <Tag
            active={true}
            key="filter-0"
            name="tomatoes"
            onClick={[Function]}
          >
            <styled.div
              active={true}
              name="tomatoes"
              onClick={[Function]}
            >
              <div
                className="sc-AxjAm caFkQs"
                name="tomatoes"
                onClick={[Function]}
              >
                <span>
                  tomatoes
                </span>
              </div>
            </styled.div>
          </Tag>
          <Tag
            active={false}
            key="filter-1"
            name="potatoes"
            onClick={[Function]}
          >
            <styled.div
              active={false}
              name="potatoes"
              onClick={[Function]}
            >
              <div
                className="sc-AxjAm YZIJc"
                name="potatoes"
                onClick={[Function]}
              >
                <span>
                  potatoes
                </span>
              </div>
            </styled.div>
          </Tag>
          <Tag
            active={true}
            key="filter-2"
            name="garlic"
            onClick={[Function]}
          >
            <styled.div
              active={true}
              name="garlic"
              onClick={[Function]}
            >
              <div
                className="sc-AxjAm caFkQs"
                name="garlic"
                onClick={[Function]}
              >
                <span>
                  garlic
                </span>
              </div>
            </styled.div>
          </Tag>
          <Tag
            active={false}
            key="filter-3"
            name="spinach"
            onClick={[Function]}
          >
            <styled.div
              active={false}
              name="spinach"
              onClick={[Function]}
            >
              <div
                className="sc-AxjAm YZIJc"
                name="spinach"
                onClick={[Function]}
              >
                <span>
                  spinach
                </span>
              </div>
            </styled.div>
          </Tag>
        </div>
      </styled.div>
    </div>
  </styled.div>
</Recommended>
`;
